### PR TITLE
vim: msg_id is a string

### PR DESCRIPTION
**Pull request by robertgzr**
my msg_id's look like: b85, d39, ae0, etc.
and I get this error when using the vim plugin:

	Error: cannot find maildir message by short hash "0" at "/path/to/my/INBOX"
	Caused by:
	0: cannot find maildir message id from short hash "0"
	1: the short hash "0" matches more than one hash: 030598120934103c456ce08338886728, 06edb10a55efb89de45d8560aee33c8e

Signed-off-by: Robert Günzler <r@gnzler.io>

### CODE REVIEW

File: vim/autoload/himalaya/msg.vim [hunk 1]
```diff
@@ -375,7 +376,7 @@ function! s:bufwidth()
 endfunction
 
 function! s:get_msg_id(line)
-    return matchstr(a:line, '[0-9]*')
+  return matchstr(a:line, '[0-9a-z]*')
```
Comment (soywod) [inline vim/autoload/himalaya/msg.vim:379]: In fact it should be `[0-9a-fA-F]*`, see https://datatracker.ietf.org/doc/html/rfc4122#section-3.

**Review by soywod — CHANGES_REQUESTED** [review]
Thanks a lot for you contribution! Indeed, Maildir message ids are UUID strings and the Vim plugin was not up-to-date. Did you try with a regular IMAP account to see if it is backward compatible? I just left one comment, we can merge straight after.

**Comment by robertgzr** [comment]
> Did you try with a regular IMAP account to see if your patch is backward compatible?

just did. works beautifully :)

**Review by soywod — APPROVED** [review]

**Comment by soywod** [comment]
Thanks a lot :pray:

**Pull request merged**